openmc/universe.py

```diff
@@ -678,7 +678,7 @@ def filename(self):
 
     @filename.setter
     def filename(self, val):
-        cv.check_type('DAGMC filename', val, str)
+        cv.check_type('DAGMC filename', val, (Path, str))
```

Comment:
Thanks for adding in Path as an option. It is super useful to accept Pathlike types. By the way sometimes I see this as ```PathLike``` not ```Path``` elsewhere